presto-orc/src/main/java/com/facebook/presto/orc/reader/BooleanStreamReader.java

```diff
@@ -96,6 +96,13 @@ public Block readBlock(Type type)
             }
         }
 
+        if (dataStream == null && presentStream != null) {
```

Comment:
Can you also port https://github.com/prestosql/presto/pull/229/commits/6a9eca0c8382257a0cd004241b1151d4de6c4079 which is the benchmark to show the improvement to this commit as part of this commit (Short circuit block creation for null data streams in orc)?

Reply:
Definitely, I still have a few related commits to port and this is one of them. Thanks!

Reply:
Done.